community/cypher/cypher-compiler-3.0/src/main/scala/org/neo4j/cypher/internal/compiler/v3_0/planner/PlannerQuery.scala

```diff
@@ -188,6 +188,100 @@ object PlannerQuery {
     val patternRelIds = patternRels.flatMap(_.coveredIds)
     patternNodeIds ++ patternRelIds
   }
+
+  /*
+  * The first reading leaf node is always stable. However for every preceding leaf node
+  * we must make sure there are no updates in this planner query that will match any of the reads.
+  * If so we must make that read a RepeatableRead.
+  */
+  // TODO: Move this into separate file?
+  def conflicts(plan: LogicalPlan, plannerQuery: PlannerQuery, fistLeafIsStable: Boolean = false): Boolean = {
+    val leaves = plan.leaves.collect {
+      case n: NodeLogicalLeafPlan => n.idName
+    }
+
+    val (stableLeaf, unstableLeaves) =
+      if (leaves.isEmpty)
+        (None, Seq.empty)
+      else if (fistLeafIsStable)
+        (Some(leaves.head), leaves.tail)
+      else
+        (None, leaves)
+
+    @tailrec
+    def conflicts(plannerQuery: PlannerQuery, next: PlannerQuery): Boolean = {
+      val conflict = if (next.updateGraph.isEmpty) false
+      else if (!fistLeafIsStable)
+        next.updateGraph overlaps plannerQuery.queryGraph
```

Comment:
We could refactor this to be a bit more pretty, now it got a bit smashed together